packages/nx/src/command-line/release/release.ts

```diff
@@ -230,23 +248,24 @@ export function createAPI(overrideReleaseConfig: NxReleaseConfiguration) {
       }
     }
 
-    const shouldCreateWorkspaceRelease = shouldCreateGitHubRelease(
-      nxReleaseConfig.changelog.workspaceChangelog
-    );
-
     let hasPushedChanges = false;
-    let latestCommit: string | undefined;
-
-    if (shouldCreateWorkspaceRelease && changelogResult.workspaceChangelog) {
-      output.logSingleLine(`Pushing to git remote`);
-
-      // Before we can create/update the release we need to ensure the commit exists on the remote
+    if (shouldPush) {
+      output.logSingleLine(`Pushing to git remote "origin"`);
       await gitPush({
         dryRun: args.dryRun,
         verbose: args.verbose,
       });
-
       hasPushedChanges = true;
+    }
+
+    let latestCommit: string | undefined;
+
+    if (shouldCreateWorkspaceRelease && changelogResult.workspaceChangelog) {
+      if (!hasPushedChanges) {
+        throw new Error(
```

Comment:
The error is here is a safety fallback, but it shouldn't be possible to reach this point because the invalid combination should be caught upfront during config validation